api/v1alpha3/haproxyloadbalancer_types.go

```diff
@@ -37,6 +38,86 @@ type HAProxyLoadBalancerSpec struct {
 	// deployed VM.
 	// +optional
 	User *SSHUser `json:"user,omitempty"`
+
+	// ExtraServices is additional services that will proxied to.
+	// +optional
+	ExtraServices []ExtraService `json:"extraServices,omitempty"`
+}
+
+// ExtraServices defines additional services that will be proxied by HAProxyLoadBalancer.
+type ExtraService struct {
+	// The name of this service
+	Name string `json:"name"`
+
+	// The list of ports that are exposed by this service.
+	Ports []ExtraServicePort `json:"ports"`
+
+	// Label selector for backend servers. Machines matching
+	// the labels will be selected as targets for the backend servers
+	// +optional
+	MachineSelector map[string]string `json:"machineSelector,omitempty"`
+
+	// HealthCheck options
+	// +optional
+	HealthCheck *ExtraServiceHealthCheck `json:"healthCheck,omitempty"`
+}
+
+// ExtraServicePort defines the frontend and backend ports used by HAProxyLoadBalancer
+type ExtraServicePort struct {
+	// Port is the haproxy port that will bind to the service
+	Port uint32 `json:"port"`
+
+	// TargetPort is the port that the service is listening on. If not specified use Port.
+	// +optional
+	TargetPort uint32 `json:"targetPort,omitempty"`
+}
+
+// ExtraServiceHealthCheck defines the health check
+type ExtraServiceHealthCheck struct {
+	// HealthCheckOption used to perform health check
+	HealthCheckOption `json:",inline"`
+}
+
+// HealthCheckOption defines the different health checking options available
+type HealthCheckOption struct {
+	// HTTP health check option
+	// +optional
+	HTTPCheck *HTTPCheck `json:"httpCheck,omitempty"`
+}
+
+// HTTPCheck defines HTTP health check
+type HTTPCheck struct {
+	// HTTP method to use for health check
+	// +optional
+	Method string `json:"method,omitempty"`
+
+	// URI to use for health check
+	// +optional
+	URI string `json:"uri,omitempty"`
+
+	// HTTP version to use for health check
+	// +optional
+	Version string `json:"version,omitempty"`
+
+	// HTTP scheme to use (HTTP or HTTPS)
+	// +optional
+	Scheme corev1.URIScheme `json:"scheme,omitempty"`
```

Comment:
I am wondering if should have an option for turning off ssl checks for instances where self signed certs are used.

Reply:
Verify is now configurable. And verify false is the default.